kie-server-parent/kie-server-controller-plugin/README.md

```diff
@@ -0,0 +1,484 @@
+# Kie Controller Maven Plugin
+
+Maven plugin to deploy k-jar container on remote managed kie-server cluster.
+
+Default goalPrefix: __kie-ctrl__.
+Actual goals are:
+1. deploy
+2. dispose
+3. get-server-templates
+5. get-containers
+6. get-container
+7. create-container
+8. delete-container
+9. start-container
+10. stop-container
+
+## Plugin configuration
+
+On pom.xml of local project add plugin: 
+
+```xml
+<build>
+  <plugins>
+    ...
+    <plugin>
+        <groupId>org.kie.server</groupId>
+        <artifactId>kie-server-controller-plugin</artifactId>
+        <version>8.0.0-SNAPSHOT</version>
+        <configuration>
+          <controllerUsername>${controller.username}</controllerUsername>
+          <controllerPassword>${controller.password}</controllerPassword>
+          <username>${kieserver.username}</username>
+          <password>${kieserver.password}</password>
+          <templateId>process-server</templateId>
+        </configuration>
+    </plugin>
+        
+  </plugins>
+</build>
+```
+
+## Managed Deploy 
+
+Managed deploy is performed by a process server central controller (business central or kie-wb).
+Process instances must have been registered to the controller using the system properties that allow to form a cluster.
+A cluster of managed process server is called a __server template__.
+
+Process server controller can manage more server template.
+A managed deploy consists to __add and start a new container to a server template__.
+Process server instances member of a server template will have all the same containers with the same configurations.
+ 
+## Goal: get-server-templates
+
+This is an idempotent/query goal to verify server template topologies and configurations. 
+
+The only mandatory parameters are __controllerUsername__ and __controllerPassword__, which are the Basic Authentication credential of remote controller.
+The controller user principal must have __rest-all__ role.
```

Comment:
I don't think this is valid, controller does is not protected by any roles as far as I know